builtin/providers/aws/structure.go

```diff
@@ -207,3 +207,21 @@ func expandStringList(configured []interface{}) []string {
 	}
 	return vs
 }
+
+// Flattens an array of DBSecurityGroups into a []string
+func flattenEC2SecurityGroupStatuses(list []rds.EC2SecurityGroup) []string {
+	result := make([]string, 0, len(list))
+	for _, i := range list {
+		result = append(result, *i.Status)
+	}
+	return result
+}
+
+// Flattens an array of IPRanges into a []string
+func flattenIPRangeStatuses(list []rds.IPRange) []string {
+	result := make([]string, 0, len(list))
+	for _, i := range list {
+		result = append(result, *i.Status)
+	}
+	return result
+}
```

Comment:
I would probably just range over them and add to the `statuses` array in the `..AuthorizeRule` func instead of creating two arrays and then joining those together. Don't think it's complex enough to warrant any reflection, though I may be wrong.